### PR TITLE
simple update

### PR DESCRIPTION
**Pull request by Domaman202**

### CODE REVIEW

File: src/main/java/de/mari_023/fabric/ae2wtlib/mixin/ContainerTypeBuilderMixin.java [hunk 1]
```diff
@@ -37,14 +40,18 @@ public void serverPacketData(PlayerEntity player, ContainerLocator locator, Call
...
-
-        if(hostInterface.isAssignableFrom(WPTGuiObject.class) && currentTerminal.equals("pattern"))
-            cir.setReturnValue(hostInterface.cast(new WPTGuiObject((ItemWT) it.getItem(), it, player, locator.getItemIndex())));
+        I result = terminalCheck(WCTGuiObject.class, currentTerminal, "crafting", it, player, locator);
+        if (result == null)
+            result = terminalCheck(WPTGuiObject.class, currentTerminal, "pattern", it, player, locator);
+        if (result == null)
+            result = terminalCheck(WITGuiObject.class, currentTerminal, "interface", it, player, locator);
+        cir.setReturnValue(result);
+    }
```
Comment (Mari023) [inline src/main/java/de/mari_023/fabric/ae2wtlib/mixin/ContainerTypeBuilderMixin.java:49]: I like this solution!
there is just one problem: it doesn't work, and I don't know why.
I do however have a solution to this which I think is even better which will push soon

**Review by Mari023 — CHANGES_REQUESTED** [review]

File: src/main/java/de/mari_023/fabric/ae2wtlib/mixin/ContainerTypeBuilderMixin.java (hunk 1 above)
Comment (Mari023) [reply src/main/java/de/mari_023/fabric/ae2wtlib/mixin/ContainerTypeBuilderMixin.java:49]: also, this would probably have broken every other ae2 screen

Comment (Mari023) [reply src/main/java/de/mari_023/fabric/ae2wtlib/mixin/ContainerTypeBuilderMixin.java:49]: (will be resolved by [#1](https://github.com/Domaman202/AE2WirelessTerminalLibrary/pull/1))

File: src/main/java/de/mari_023/fabric/ae2wtlib/mixin/PlayerInventoryInsertStack.java [hunk 2]
```diff
@@ -29,14 +29,16 @@ public void insertStackInME(ItemStack stack, CallbackInfoReturnable<Boolean> cir
...
-        } catch(NullPointerException ignored) {}
+        if(ItemMagnetCard.isPickupME(terminal) && CTHandler.inRange()) {
+            if (CTHandler.getItemStorageChannel() == null) return;
+            try {
+                IAEItemStack leftover = CTHandler.getItemStorageChannel().injectItems(AEItemStack.fromItemStack(stack), Actionable.MODULATE, new PlayerSource(player, (IActionHost) CTHandler.getSecurityStation()));
+                if (leftover == null || leftover.createItemStack().isEmpty()) {
+                    stack.setCount(0);
+                    cir.setReturnValue(true);
+                } else stack.setCount(leftover.createItemStack().getCount());
+            } catch (NullPointerException ignored) { // TODO: REPLACE THIS TO NULL SAFE BLOCK!
```
[the remark this answers is not in the document]
Comment (Domaman202) [reply src/main/java/de/mari_023/fabric/ae2wtlib/mixin/PlayerInventoryInsertStack.java:40]: lol

File: src/main/java/de/mari_023/fabric/ae2wtlib/rei/RecipeTransferHandler.java [hunk 3]
```diff
@@ -24,7 +25,7 @@
     }
 
     @Override
-    public AutoTransferHandler.Result handle(AutoTransferHandler.Context context) {
+    public AutoTransferHandler.@NotNull Result handle(AutoTransferHandler.Context context) {
```
[the remark this answers is not in the document]
Comment (Domaman202) [reply src/main/java/de/mari_023/fabric/ae2wtlib/rei/RecipeTransferHandler.java:28]: intellij idea complain.

File: src/main/java/de/mari_023/fabric/ae2wtlib/mixin/ContainerTypeBuilderMixin.java [hunk 4]
```diff
@@ -1,19 +1,17 @@
...
-import de.mari_023.fabric.ae2wtlib.wit.WITGuiObject;
-import de.mari_023.fabric.ae2wtlib.wpt.WPTGuiObject;
 import de.mari_023.fabric.ae2wtlib.wut.WUTHandler;
 import dev.emi.trinkets.api.TrinketsApi;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.item.ItemStack;
 import org.spongepowered.asm.mixin.Final;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
```
Comment (Mari023) [inline src/main/java/de/mari_023/fabric/ae2wtlib/mixin/ContainerTypeBuilderMixin.java:14]: import is now unused and can be removed

**Review by Mari023 — APPROVED** [review]

**Pull request merged**